src/hotspot/share/gc/shared/spaceDecorator.cpp

```diff
@@ -23,64 +23,11 @@
  */
 
 #include "precompiled.hpp"
-#include "gc/parallel/mutableSpace.hpp"
-#include "gc/shared/space.hpp"
-#include "gc/shared/spaceDecorator.inline.hpp"
-#include "logging/log.hpp"
+#include "gc/shared/spaceDecorator.hpp"
 #include "utilities/copy.hpp"
 
-// Catch-all file for utility classes
-
 #ifndef PRODUCT
 
-// Returns true is the location q matches the mangling
-// pattern.
-bool SpaceMangler::is_mangled(HeapWord* q) {
-  // This test loses precision but is good enough
-  return badHeapWord == (max_juint & reinterpret_cast<uintptr_t>(*q));
-}
-
-
-void SpaceMangler::set_top_for_allocations(HeapWord* v)  {
-  if (v < end()) {
-    assert(!CheckZapUnusedHeapArea || is_mangled(v),
-      "The high water mark is not mangled");
-  }
-  _top_for_allocations = v;
-}
-
-// Mangle only the unused space that has not previously
-// been mangled and that has not been allocated since being
-// mangled.
-void SpaceMangler::mangle_unused_area() {
-  assert(ZapUnusedHeapArea, "Mangling should not be in use");
-  // Mangle between top and the high water mark.  Safeguard
-  // against the space changing since top_for_allocations was
-  // set.
-  HeapWord* mangled_end = MIN2(top_for_allocations(), end());
-  if (top() < mangled_end) {
-    MemRegion mangle_mr(top(), mangled_end);
-    SpaceMangler::mangle_region(mangle_mr);
-    // Light weight check of mangling.
-    check_mangled_unused_area(end());
-  }
-  // Complete check of unused area which is functional when
-  // DEBUG_MANGLING is defined.
-  check_mangled_unused_area_complete();
-}
-
-// A complete mangle is expected in the
-// exceptional case where top_for_allocations is not
-// properly tracking the high water mark for mangling.
-// This can be the case when to-space is being used for
-// scratch space during a mark-sweep-compact.  See
-// contribute_scratch().
-void SpaceMangler::mangle_unused_area_complete() {
-  assert(ZapUnusedHeapArea, "Mangling should not be in use");
-  MemRegion mangle_mr(top(), end());
-  SpaceMangler::mangle_region(mangle_mr);
-}
-
 // Simply mangle the MemRegion mr.
 void SpaceMangler::mangle_region(MemRegion mr) {
```

Comment:
SpaceMangler::mangle_region should be debug-only rather than not-product, both conceptually, and
because it has no effect in non-debug builds.